File: Src/java/tools/xsd-to-modelinfo/src/main/java/org/cqframework/cql/tools/xsd2modelinfo/ModelImporterMapperValue.java
package org.cqframework.cql.tools.xsd2modelinfo;

import java.util.HashMap;
import java.util.Map;

public class ModelImporterMapperValue {
    public enum Relationship {
        RETYPE,
        EXTEND
    }

    private final String targetSystemClass;
    private final Relationship relationship;
    private final Map<String, String> targetClassElementMap;

    public ModelImporterMapperValue(String targetSystemClass, Relationship relationship) {
        this.targetSystemClass = targetSystemClass;
        this.relationship = relationship;
        this.targetClassElementMap = new HashMap<>();
    }

    public String getTargetSystemClass() {
        return targetSystemClass;
    }

    public Relationship getRelationship() {
        return relationship;
    }

    public Map<String, String> getTargetClassElementMap() {
        return targetClassElementMap;
    }

    public void addClassElementMapping(String element, String targetElement) {
        targetClassElementMap.put(element, targetElement);
    }

    public static ModelImporterMapperValue newRetype(String targetSystemClass) {
        return new ModelImporterMapperValue(targetSystemClass, Relationship.RETYPE);
    }

    public static ModelImporterMapperValue newExtend(String targetSystemClass) {
        return new ModelImporterMapperValue(targetSystemClass, Relationship.EXTEND);
    }
}

File: Src/java/tools/xsd-to-modelinfo/src/main/kotlin/org/cqframework/cql/tools/xsd2modelinfo/ModelImporterMapperValue.kt
package org.cqframework.cql.tools.xsd2modelinfo

class ModelImporterMapperValue(@JvmField val targetSystemClass: String?, @JvmField val relationship: Relationship?) {
    enum class Relationship {
        RETYPE,
        EXTEND
    }

    val targetClassElementMap: MutableMap<String?, String?>

    init {
        this.targetClassElementMap = HashMap<String?, String?>()
    }

    fun addClassElementMapping(element: String?, targetElement: String?) {
        targetClassElementMap[element] = targetElement
    }

    companion object {
        fun newRetype(targetSystemClass: String?): ModelImporterMapperValue {
            return ModelImporterMapperValue(targetSystemClass, Relationship.RETYPE)
        }

        fun newExtend(targetSystemClass: String?): ModelImporterMapperValue {
            return ModelImporterMapperValue(targetSystemClass, Relationship.EXTEND)
        }
    }
}
